### DISCUSSION TITLE
Any way to trigger a library scan from the CLI/API?

### DISCUSSION THREAD

**Discussion by neingeist** [advplyr/audiobookshelf#1012] [Q&A] [answered]
I'm looking for a way to trigger a library scan from the CLI/API - how do I do it? Next question would be if there are any API tokens (that could be restricted even, but this does not matter much to me currently.)

Also a related side note: I love Audiobookshelf! Thanks for developing it!

**Comment by advplyr** [comment, marked as answer]
Hey there is a full API to do everything using the JWT shown on the users page. Unfortunately this isn't documented well yet but for a few endpoints that are probably a little out of date now.

https://api.audiobookshelf.org/

The goal is to have full API docs there but you can also see the api endpoints in the code here: https://github.com/advplyr/audiobookshelf/blob/master/server/routers/ApiRouter.js#L56

**Reply by neingeist** [reply]
Thanks for your reply! One question: what is "the JWT shown on the users page"?

I did a few experiments a while back, and I could issue a refresh by:

~~~
curl -H 'Authorization: Bearer <token>' \
    'https://audiobooks.example.com/api/libraries/lib_e129zwocd6urxx9dyom/scan'  
~~~
where I got the `<token>` from observing what Firefox sends.

**Reply by neingeist** [reply]
> One question: what is "the JWT shown on the users page"?

Ah, that must be it: Navigating to `/config/users/<username>`, there is now an API token field.
